main.tf

```diff
@@ -253,15 +253,22 @@ resource "aws_lambda_function" "this" {
     variables = {
       AWS_REGIONS               = var.aws_regions == null ? data.aws_region.current.name : join(", ", var.aws_regions)
       SCHEDULE_ACTION           = var.schedule_action
-      TAG_KEY                   = var.resources_tag["key"]
-      TAG_VALUE                 = var.resources_tag["value"]
+      TAG_KEY                   = var.resources_tag[ "key" ]
+      TAG_VALUE                 = var.resources_tag[ "value" ]
       EC2_SCHEDULE              = tostring(var.ec2_schedule)
       RDS_SCHEDULE              = tostring(var.rds_schedule)
       AUTOSCALING_SCHEDULE      = tostring(var.autoscaling_schedule)
       CLOUDWATCH_ALARM_SCHEDULE = tostring(var.cloudwatch_alarm_schedule)
     }
   }
 
+  lifecycle {
+    ignore_changes = [
+      filename,
+      last_modified,
+      source_code_hash,
+    ]
+  }
```

Comment:
@mixe3y I don't understand the usefulness of this block.
Why did you add this block ?

Reply:
Hi @diodonfrost !

That was not intended to be in the PR yet.

However I found a weird issue, that's what I'm trying to fix.

I use terraform in 2 environments. On my own notebook and on a server (shared with my colleagues).
When I apply the module on my computer and I apply it on the remote it wants change two things (without changing anything code related). 

See the code snippet below.

```
  # module.stop_ec2_instance.aws_lambda_function.this will be updated in-place
  ~ resource "aws_lambda_function" "this" {
        arn                            = "arn:aws:lambda:ap-southeast-2:XXX:function:ec2_stop"
        filename                       = ".terraform/modules/stop_ec2_instance/aws-stop-start-resources.zip"
        function_name                  = "ec2_stop"
        handler                        = "scheduler.main.lambda_handler"
        id                             = "ec2_stop"
        invoke_arn                     = "arn:aws:apigateway:ap-southeast-2:lambda:path/2015-03-31/functions/arn:aws:lambda:ap-southeast-2:XXX:function:ec2_stop/invocations"
      ~ last_modified                  = "2020-11-06T15:26:53.492+0000" -> (known after apply)
        layers                         = []
        memory_size                    = 128
        publish                        = false
        qualified_arn                  = "arn:aws:lambda:ap-southeast-2:XXX:function:ec2_stop:$LATEST"
        reserved_concurrent_executions = -1
        role                           = "arn:aws:iam::XXX:role/ec2_stop-scheduler-lambda"
        runtime                        = "python3.7"
      ~ source_code_hash               = "Dt7sCTtSHLP5GPoLxbmMw5xUT0kN6AfufWDdGdfeVDk=" -> "+cbvC8YOi1g/+GZkHQqNYHzxrHtMJj+030zGaXaFHH8="
        source_code_size               = 7012
        tags                           = {}
        timeout                        = 600
        version                        = "$LATEST"

        environment {
            variables = {
                "AUTOSCALING_SCHEDULE"      = "false"
                "AWS_REGIONS"               = "ap-southeast-2"
                "CLOUDWATCH_ALARM_SCHEDULE" = "false"
                "EC2_SCHEDULE"              = "true"
                "RDS_SCHEDULE"              = "false"
                "SCHEDULE_ACTION"           = "stop"
                "TAG_KEY"                   = "tostop"
                "TAG_VALUE"                 = "true"
            }
        }

        tracing_config {
            mode = "PassThrough"
        }
    }

Plan: 0 to add, 2 to change, 0 to destroy.

```

Reply:
I had never thought of that.
I have the same issue when run terraform apply on two different operating systems.

This problem can be mitigate by removing this line:
```
source_code_hash = data.archive_file.this.output_base64sha256
```

Reply:
Good catch!